lib/sprinkle_dns/client.rb

```diff
@@ -89,32 +84,59 @@ def sprinkle!
         return [existing_hosted_zones, nil]
       end
 
+      hosted_zones = (existing_hosted_zones + missing_hosted_zones)
+
       unless @config.force?
-        changes = existing_hosted_zones.map{|h| SprinkleDNS::EntryPolicyService.new(h, @config)}.collect{|eps| eps.entries_to_change}.sum
-        puts
-        print "#{changes} changes to make. Continue? (y/N)"
+        changes = hosted_zones.map{|h| SprinkleDNS::EntryPolicyService.new(h, @config)}.collect{|eps| eps.entries_to_change}.sum
+
+        messages = []
+
+        messages << "#{missing_hosted_zones.size} hosted-zone(s) to create" if missing_hosted_zones.any?
+        messages << "#{changes} change(s) to make" if changes > 0
+        print messages.join(' and ').concat(". Continue? (y/N)")
+
         case gets.strip
         when 'y', 'Y'
           # continue
         else
           puts ".. exiting!"
-          return [existing_hosted_zones, nil]
+          return [hosted_zones, nil]
         end
       end
 
-      change_requests = @dns_provider.change_hosted_zones(existing_hosted_zones, @config)
       progress_printer = if @config.interactive_progress?
-        SprinkleDNS::CLI::InteractiveChangeRequestPrinter.new
+        SprinkleDNS::CLI::InteractiveChangeRequestPrinter
       else
-        SprinkleDNS::CLI::PropagatedChangeRequestPrinter.new
+        SprinkleDNS::CLI::PropagatedChangeRequestPrinter
       end
 
-      begin
-        @dns_provider.check_change_requests(change_requests)
-        progress_printer.draw(change_requests)
-      end until change_requests.all?{|cr| cr.in_sync}
+      # Create missing hosted zones
+      change_requests_hosted_zones = @dns_provider.create_hosted_zones(missing_hosted_zones)
+      if change_requests_hosted_zones.any?
+        puts
+        puts "Creating hosted zones:"
+        printer = progress_printer.new
```

Comment:
(minor) imo it's best to use dependency injection from the upper layer here, and have an instance ready, instead of taking a class

Reply:
Fixed in https://github.com/gfish/sprinkle_dns/pull/14/commits/7e3bff2c1fda2e7fee98b8224131887d52ef304a